src/screens/VaultScreen.js

```diff
@@ -13,11 +14,305 @@ export default class VaultScreen extends React.Component {
+      'source-sans-pro-regular': require('../.././assets/fonts/SourceSansPro-Regular.ttf'),
+      'source-sans-pro-semibold': require('../.././assets/fonts/SourceSansPro-SemiBold.ttf'),
+      'source-sans-pro-semibold-italic': require('../.././assets/fonts/SourceSansPro-SemiBoldItalic.ttf'),
+    });
+    this.setState({fontLoaded: true});
+  }
+  constructor(props) {
+    super(props);
+    this.state = {
+      titleName: "Resources",
+      searchText: '',
+      filter: [],
+      names: [
+        {name: 'CLEAR', selected:false, id: 1},
+        {name: 'FINANCE', selected:true, id: 2},
+        {name: 'HOUSING', selected:true, id: 3},
+        {name: 'EMPLOYMENT', selected:true, id: 4},
+        {name: 'LEGAL', selected:true, id: 5},
+        {name: 'MENTAL HEALTH', selected:true, id: 6},
+      ],
+      fontLoaded: false,
+      resources: [
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 1,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 2,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 3,
+        }
+      ]
+    };
+  }
+  updateFilter(item_id, new_state) {
+    var names_arr = this.state.names.slice()
+    for(var i = 0; i < names_arr.length; i++) {
+      if(names_arr[i].id == item_id) {
+        names_arr[i].selected = new_state;
+      }
+    };
+    this.setState({names:names_arr});
+  }
+  setOpposite(item_id) {
+    for(var i = 0; i < this.state.names.length; i++) {
+      if(this.state.names[i].id == item_id) {
+        this.updateFilter(item_id, !this.state.names[i].selected);
+      }
+    };
+  }
+  falseState(name, item_id) {
+    if(name=='Clear') {
+      for(var i = 1; i < this.state.names.length; i++) {
+        this.updateFilter(this.state.names[i].id, false)
+      }
+    } else {
+      this.setOpposite(item_id);
+    }
+  }
+  filterScroller() {
+    return this.state.names.map((item) => {
+      if (item.selected==false) {
+        return (
+          <TouchableHighlight key = {item.id} style={styles.item} onPress={() => {this.falseState(item.name, item.id);}}>
```

Comment:
I have a feeling that this might trigger unexpected errors later on since you're using an unbound function. Whenever you pass in a method as a props to another component make sure you bind it in the constructor. For example if I call this method in `onPress` "callback", then I would do this to bind it:

```javascript
constructor(props) {
  // Do stuff
  this.callback = this.callback.bind(this);
}
```